### ISSUE TITLE
Asking permission non-stop

### ISSUE THREAD

**Issue by zackify** [kryptco/kryptonite-ios#71]
I have my device set to never ask and to not ask on unknown hosts. Yet as of yesterday it has been prompting approval on every ssh connection.

**Comment by zackify** [comment]
Deleted the device, repaired, and fixed the problem. Looks like there's a bug though. Wonder if anyone else has ran into this.

**Comment by agrinman** [comment]
This is very strange, what notification text did you see -- was it "Kryptonite Request"?

**Comment by zackify** [comment]
Yes it was. So I would tell it allow once. Or three hours. It would then work. And the device would be set to one of those options. Choosing "never" didn't seem to take any effect. Would keep requesting each time if I had that set. Wish I could have shown a better way to reproduce.

**Comment by agrinman** [comment]
Yes, this is a known issue (with a potentially unknown cause).`Kryptonite Request` should never actually appear, it does when the iOS NotificationServiceExtension breaks while processing a "mutable" remote notification containing the encrypted request payload. This enables Kryptonite to decrypt a login request and either auto-approve per the set policies or convert it to a request for the user to approve even when the app is *not* running in the background. If the NotificationServiceExtension breaks, it defaults to having no detailed message and ignoring device policies, asking the user to approve directly. Unfortunately, it looks like the NotificationServiceExtension has some weird/unreliable behavior how it's implemented. Restarting the app or phone might even fix these issues.

Are you running iOS 11 beta by any chance?

**Comment by zackify** [comment]
That's too bad. Nope. iOS 10.3.2

**Comment by agrinman** [comment]
Ok thanks, sorry about this! Please let me know if this starts happening again and if restarting the app fixes it at all.
